src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs

```diff
@@ -79,10 +82,30 @@ public async Task<PagedList<APIRevisionListItemModel>> GetAPIRevisionsAsync(Page
         /// Retrieve Revisions for a particular Review from the Revisions container in CosmosDb
         /// </summary>
         /// <param name="reviewId"></param> The Reviewid for which the revisions are to be retrieved
-        /// <returns></returns>
-        public async Task<IEnumerable<APIRevisionListItemModel>> GetAPIRevisionsAsync(string reviewId)
+        /// <param name="updateCache"></param> update memory cache
+        /// <param name="queryCache"></param> query cache for APIrevisions
+        public async Task<IEnumerable<APIRevisionListItemModel>> GetAPIRevisionsAsync(string reviewId, bool updateCache = false, bool queryCache = false)
         {
-            return await _apiRevisionsRepository.GetAPIRevisionsAsync(reviewId);
+            IEnumerable<APIRevisionListItemModel> apiRevisions = new List<APIRevisionListItemModel>();
+            if (queryCache)
+            {
+                _cache.TryGetValue(reviewId, out apiRevisions);
+            }
+
+            if (apiRevisions == null || !apiRevisions.Any())
+            {
+                apiRevisions = await _apiRevisionsRepository.GetAPIRevisionsAsync(reviewId);
+            }
+
+            if (updateCache && apiRevisions.Any())
+            {
+                var cacheEntryOptions = new MemoryCacheEntryOptions()
+                    .SetAbsoluteExpiration(TimeSpan.FromHours(24));
```

Comment:
This is too long. We should cache only for few mins like 10 mins. 24 hours is too long for revisions cache. This will cause to use a lot of memory for every review accessed in last 24 hours.

Reply:
We also need to invalidate/update the cache when a new revision is created/ modified like approved.

Reply:
I think it might be better to just load the latest 30 revisions in the dropdown since the performance issue is on the UI side.